community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundQueryContext.scala

```diff
@@ -456,7 +482,13 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
         val buffer = ArrayBuffer[Int]()
         nodeCursor.properties(propertyCursor)
         while (propertyCursor.next()) {
-          buffer.append(propertyCursor.propertyKey())
+          val propertyKeyId = propertyCursor.propertyKey()
+          try {
+            transactionalContext.readOperations.propertyKeyGetName(propertyKeyId)
+            buffer.append(propertyKeyId)
+          } catch {
```

Comment:
Using exceptions for this seems a bit expensive. Since this need to be handled during operation in normal conditions it would be great if we could use an augmented cursor or at least use a return value to communicate the 'not found' status (we could add a new API method for this if necessary).